engine/table/src/test/java/io/deephaven/engine/table/impl/TestPartitionBy.java

```diff
@@ -58,42 +56,44 @@ class TableMapNugget implements EvalNuggetInterface {
         @Override
         public void validate(String msg) {
             // get all the keys from the original table
-            final HashSet<Object> keys = new HashSet<>();
+            final HashSet<Object[]> keys = new HashSet<>();
 
+            // TODO-RWC: Don't use smart keys
```

Comment:
Thanks, I should have deleted that `TODO`. Also, you're definitely right; I'm sure I was testing the same partition multiple times, since it was probably doing identity hashing and equality.
Replaced it with a `HashMap<ArrayTuple, Object[]>`. Should be no more expensive than a `HashSet` which is backed by a `HashMap<TYPE, Boolean>`, and avoids having to re-create the key array later.